tonic-build/src/lib.rs

```diff
@@ -40,6 +40,21 @@
 //!         )?;
 //!    Ok(())
 //! }
+//!
+//! ### NixOS related hints
+//!
+//! On NixOS, it is better to specify the location of `PROTOC` and `PROTOC_INCLUDE` explicitly.
+//!
+//! ```bash
+//! $ export PROTOBUF_LOCATION=$(nix-env -q protobuf --out-path --no-name)
+//! $ export PROTOC=$(PROTOBUF_LOCATION)/bin/protoc
+//! $ export PROTOC_INCLUDE=$(PROTOBUF_LOCATION)/include
+//! $ cargo build
+//! ```
+//!
+//! The reason being that if `prost_build::compile_protos` fails to generate the resultant package,
+//! the failure is not obvious until the `include!(concat!(env!("OUT_DIR"), "/resultant.rs"));`
+//! fails with `No such file or directory` error.
 //! ```
```

Comment:
I feel like this one needs to be above the `NixOS` header?

Reply:
The comment about `prost_build`?

Reply:
Yeah, I mean the code block above what you added, it looks like the backticks got moved down instead of closing the code block, right?

Reply:
Oh crap, yes! Thank you. Sorry I need more coffee.